multipaz-models/src/commonMain/kotlin/org/multipaz/models/presentment/mdocPresentment.kt

```diff
@@ -104,6 +110,8 @@ internal suspend fun mdocPresentment(
                 encodedSessionTranscript!!,
             ).parse()
             for (docRequest in deviceRequest.docRequests) {
+                val usingZkRequest = docRequest.getZkSystemSpecs().isNotEmpty()
```

Comment:
Better to call this `readerRequestedZk`, we don't know yet if we can actually honor it... then move the block below that finds the circuit up here which will determine _if_ we can actually honor the request (we might not support the circuit) and put that in `usingZkRequest`...

Otherwise we might end up not using key agreement while still doing a traditional response.